### PR TITLE
refactor: combine PC host and port into a single url var

### PR DESCRIPTION
**Pull request by dkoshkin**
Wanted to hear the team's thoughts on this.
@supershal originally brought this [up in the original PR](https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/6).

This makes it simpler for clients to provide a single input field and not have to doany parsing to split the hostname and port.
It also allows us to use API validation for bad input.

### CODE REVIEW

File: api/v1alpha1/nutanix_clusterconfig_types.go [hunk 1]
```diff
@@ -65,17 +60,12 @@ func (NutanixPrismCentralEndpointSpec) VariableSchema() clusterv1.VariableSchema
...
 					MinLength:   ptr.To[int64](1),
-				},
-				"port": {
-					Description: "The port number to access the Nutanix Prism Central",
-					Type:        "integer",
-					Default:     variables.MustMarshal(PrismCentralPort),
-					Minimum:     ptr.To[int64](1),
-					Maximum:     ptr.To[int64](65535),
+					Format:      "uri",
+					Pattern:     "^https://",
```
Comment (supershal) [inline api/v1alpha1/nutanix_clusterconfig_types.go:68]: are we allowing `http://` with `insecure: true`

Comment (dkoshkin) [reply api/v1alpha1/nutanix_clusterconfig_types.go:68]: No, the schema is always `https` (the `Pattern` enforces that). `insecure: true` is only used for `TLSSkipVerify` when the serving cert was signed by a self-signed CA.

**Review by supershal — APPROVED** [review]

File: api/v1alpha1/nutanix_clusterconfig_types.go (hunk 1 above)
Comment (dkoshkin) [reply api/v1alpha1/nutanix_clusterconfig_types.go]: Totally forgot we had this package, thanks @faiq 
Fixed in https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/36/commits/dd1af816d19d6bdc1eacffed45291002aad7db7c

**Review by faiq — APPROVED** [review]

File: api/v1alpha1/nutanix_clusterconfig_types.go (hunk 1 above)
Comment (supershal) [reply api/v1alpha1/nutanix_clusterconfig_types.go]: nice.

**Pull request merged**